dd-sdk-android/src/main/kotlin/com/datadog/android/rum/webview/WebRumEventConsumer.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.webview
+
+import com.datadog.android.core.internal.persistence.DataWriter
+import com.datadog.android.core.internal.time.TimeProvider
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.rum.internal.domain.RumContext
+import com.datadog.android.rum.model.ActionEvent
+import com.datadog.android.rum.model.ErrorEvent
+import com.datadog.android.rum.model.LongTaskEvent
+import com.datadog.android.rum.model.ResourceEvent
+import com.datadog.android.rum.model.ViewEvent
+import com.google.gson.JsonObject
+import com.google.gson.JsonParseException
+import java.util.Locale.US
+import kotlin.collections.LinkedHashMap
+
+internal class WebRumEventConsumer(
+    private val dataWriter: DataWriter<Any>,
+    private val timeProvider: TimeProvider,
+    private val webRumEventMapper: WebRumEventMapper = WebRumEventMapper(),
+    private val contextProvider: WebRumEventContextProvider = WebRumEventContextProvider()
+) {
+
+    internal val offsets: LinkedHashMap<String, Long> = LinkedHashMap()
+    private var rumFeatureDisabled = false
+
+    fun consume(event: JsonObject, eventType: String) {
+        if (rumFeatureDisabled) {
+            return
+        }
+        val rumContext = contextProvider.getRumContext()
+        if (rumContext != null) {
+            val mappedEvent = map(event, eventType, rumContext)
+            if (mappedEvent != null) {
+                dataWriter.write(mappedEvent)
+            }
+        } else {
+            rumFeatureDisabled = true
+        }
```

Comment:
Instead of doing this (and losing data), maybe we can write the unmapped event. Because if native rum is disabled, they might still want to have the browser rum events